modules/tensor_mechanics/include/utils/RankFourTensor.h

```diff
@@ -160,7 +160,8 @@ class RankFourTensor
    * Rotate the tensor using
    * C_ijkl = R_im R_in R_ko R_lp C_mnop
    */
-  virtual void rotate(RealTensorValue & R);
+  template <class T>
+  void rotate(T & R);
```

Comment:
FYI: I added that change to allow rotate to take parameters of type `RealVectorValue` _and_ `RankTwoTensor`.
We could also add only these two specializations and move the implementation into the .C file.